morebits.js

```diff
@@ -4001,6 +4001,71 @@ Morebits.wikitext.page.prototype = {
 		}
 	},
 
+	/**
+	 * Smartly insert a tag atop page text but after specified templates,
+	 * such as hatnotes, short description, or deletion and protection templates.
+	 * Notably, does *not* insert a newline after the tag
+	 *
+	 * @param {string} tag - The tag to be inserted
+	 * @param {string|array} regex - Templates after which to insert tag,
+	 * given as either as a string or an array to be joined by pipes
+	 * @param {string} [flags=i] - Regex flags to apply. Optional, defaults to /i
+	 * @param {string|array} preRegex - Optional string or array to match before
+	 * any template matches (i.e. before `{{`), such as html comments
+	 *
+	 * @returns {string}
+	 */
+	insertAfterTemplates: function(tag, regex, flags, preRegex) {
+		if (!tag) {
+			return this.text;
+		}
+
+		// .length is only a property of strings and arrays so we
+		// shouldn't need to check type
+		if (!regex || !regex.length) {
+			return tag + this.text;
+		} else if (Array.isArray(regex)) {
+			regex = regex.join('|');
+		}
+
+		flags = flags || 'i';
+
+		if (!preRegex || !preRegex.length) {
+			preRegex = '';
+		} else if (Array.isArray(preRegex)) {
+			preRegex = preRegex.join('|');
+		}
+
+
+		// Regex is extra complicated to allow for templates with
+		// parameters and to handle whitespace properly
+		return this.text.replace(
```

Comment:
```suggestion
		return this.text = this.text.replace(
```
This should at least modify `this.text` at the least. 

I'm not sure whether it'd better to return the modified text or just `this` (which would make the method chainable). Presently, the other Morebits.wikitext methods don't return anything so changing them all to  return `this` is what would make sense IMO, so that it can be used like `text = wikipage.insertAfterTemplates(...).getText()`.

Reply:
No, that makes sense, and chaining would be ideal.  I can toss in some `return this`-es to the others, IIRC most of the uses are simple.

Reply:
Unused on enwiki, so added here.